test/fixtures/es-modules/cjs-exports.mjs

```diff
@@ -0,0 +1,35 @@
+import { strictEqual, deepEqual } from 'assert';
+
+import m, { isObject, π, z } from './exports-cases.js';
+import * as ns from './exports-cases.js';
+
+deepEqual(Object.keys(ns), ['default', 'isObject', 'z', 'π']);
+strictEqual(typeof isObject, 'undefined');
+strictEqual(π, 'yes');
+strictEqual(z, 'yes');
+strictEqual(typeof m.isObject, 'undefined');
+strictEqual(m.π, 'yes');
+strictEqual(m.z, 'yes');
+
+import m2, { __esModule as __esModule2, name as name2 } from './exports-cases2.js';
+import * as ns2 from './exports-cases2.js';
+
+strictEqual(__esModule2, true);
+strictEqual(name2, 'name');
+strictEqual(m2, 'the default');
+strictEqual(ns2.__esModule, true);
```

Comment:
This feels like a property that maybe should be hidden. It's a property that will be exported from transpiled CJS files but won't exist as a binding in the original ESM. So at best it's a really confusing feature test:

```js
import * as dep from 'dep';
const depIsESModule = !dep.__esModule; // wat?
```

At worst it's a binding that can be imported until `dep` enables `exports` and exposes native ESM.

Reply:
Breadley had some reasons for wanting this in - I believe for allowing interop like:

```js
import * as dep from 'dep';
const depValue = dep.__esModule ? dep : dep.default;
```

@bmeck is this still something you see as a requirement from your side?

Reply:
I have no requirement on it, but it feels odd to strip out potentially useful info

Reply:
I think we should keep it actually as it allows useful interop patterns.

Reply:
Speaking with @jkrems further about this he indicated he would be ok with this. Jan, if you want to reopen this discussion at any time just let me know.